cognite/client/_api/assets.py

```diff
@@ -412,32 +444,137 @@ def create(self, asset: Union[Asset, Sequence[Asset]]) -> Union[Asset, AssetList
         utils._auxiliary.assert_type(asset, "asset", [Asset, Sequence])
         return self._create_multiple(list_cls=AssetList, resource_cls=Asset, items=asset)
 
-    def create_hierarchy(self, assets: Sequence[Asset]) -> AssetList:
-        """Create an asset hierarchy.
+    def create_hierarchy(
+        self,
+        assets: Union[Sequence[Asset], AssetHierarchy],
+        *,
+        upsert: bool = False,
+        upsert_mode: UpsertOptions = "patch",
```

Comment:
Let's be consistent with the naming
```suggestion
        upsert_mode: UpsertMode = "patch",
```

Reply:
Disagree, `upsert_mode` is a selection of one of the `UpsertOptions`.

Reply:
Huh, weird, to me, it is literally (pun intended) just a collection of (specifically) allowed values. Not sure I follow the `Animal` example since a `Literal` allows wild combinations of types:
```py
# From PEP586
Literal[-4, "foo", b"foo", u"foo", True, Color.RED, None]
```
I ended up just spelling it out as `Literal["replace", "patch"]` in the type hints. I did the same for `OnErrorOptions = Literal["ignore", "warn", "raise"]` in `cognite/client/data_classes/assets.py`.

Reply:
...and yes, I would 10/10 times name that `AnimalOptions` 😜